### ISSUE TITLE
Make path structure inference more configurable

### ISSUE THREAD

**Issue by aburgel** [opticdev/optic#2337]
**Describe the enhancement**
There's a [constant called `reservedPatterns`](https://github.com/opticdev/optic/blob/d21c8eced0c0b67d620f0994b63468229a3aeedd/projects/optic/src/commands/capture/operations/infer-path-structure.ts#L373) that is a list of regexps to prevent certain path segments from being identified as resource names.

It would be really great if we could make that configurable. Or even just defining path prefixes as a start. Our API paths look something like `/api/public/<resource>/<resource_id>`. Right now, optic thinks `public` is the name of a resource and then the rest of path segments get misinterpreted.

I'm working around this by modifying `reservedPatterns` directly in a fork, but it would be much easier to have something like `x-optic-path-ignore` so this can be configured.

**Comment by acunniffe** [comment]
Hey @aburgel -- nice work finding that constant and changing some of the priors used when inferring paths. 

As long as all your routes start with `/api/public/` you should be able to set a `url` base path in `.servers` that Optic learns: 
```
servers:
  - url: http://localhost:8000/api/public
  - url: http://prod.yourapi.com/api/public
```
One consequence of using a base path is that every operation Optic documents will have a path relative to the base path. So you'd get `/<resource>/<resource_id>` at the top level, not `/api/public/<resource>/<resource_id>`. 

When Optic works with your spec in CI or in the Cloud product it will automatically join the two paths and so will most other OpenAPI tools. 

Do you think that can work for your use case?

**Comment by aburgel** [comment]
This doesn't seem to work.

> As long as all your routes start with `/api/public/`

This isn't the case for me. I do have other routes that I do not want to be added to the schema. I have added them to the ignore list, but that doesn't seem to help.

Any other suggestions?

**Comment by acunniffe** [comment]
We'll think about the right way to expose something like `reservedPatterns` to allow teams to set resource names. We always expected the first time you document an API users might need to do some of the more complex paths manually. Once they're learned or written manually, Optic will always be able to match the traffic to them going forward. 

Maybe while we're considering more options here, you could add a few known paths to the OpenAPI file manually to guide the inference. The more paths you document, the better it should get at inferring. 

```
paths: 
  /api/public ...
  /api/private:...
```

If you want to set up time to do it together and talk through what would make it easier feel free to set up time here https://calendly.com/optic-onboarding/meeting-with-aidan-from-optic-clone. I don't have enough information about your API structure to suggest much more.

**Comment by niclim** [comment]
Hi @aburgel - I just released a new version `0.50.17` which improves how Optic handles prefixes and should now work as we've described above.

That is - if you add a route `/api/public` and `/api/public/<resource>/<resource_id>` to your OpenAPI file (one route should be enough), it should infer that the `/api/public/<resource>` route is not a variable

E.g.
```
paths:
  /api/public/users/{userId}: ...
```

Then when you run capture (`optic capture openapi.yml --update=interactive`) with other paths (say for example `/api/public/organizations/some-org-id`) it should infer that the path is `/api/public/organizations/{orgId}` 

Let me know if this works for you - thanks!

**Comment by niclim** [comment]
I'm going to close this out first, but if this doesn't work for you or there are any other issues feel free to respond / reopen.

Thank you!
